### ISSUE TITLE
predictions for multi class training

### ISSUE THREAD

**Issue by run2** [benanne/Lasagne#96]
Hi
I am doing a multiclass training and prediction with a network which looks like this
net3 = NeuralNet(
    layers=[
        ('input', layers.InputLayer),
        ('conv1', Conv2DLayer),
        ('pool1', MaxPool2DLayer),
        ('dropout1', layers.DropoutLayer),  # !
        ('hidden1', layers.DenseLayer),
        ('output', layers.DenseLayer),
        ],
    batch_iterator=BatchIterator(20,True),
    input_shape=(20, 1, 96, 96),
    conv1_num_filters=32, conv1_filter_size=(6, 6), pool1_ds=(2, 2),
    dropout1_p=0.1,  # !
    hidden1_num_units=100, hidden1_nonlinearity=sigmoid,
    output_num_units=20,output_nonlinearity=sigmoid,
    update_learning_rate=0.01,
    update_momentum=0.9,
    use_label_encoder=True,
    regression=False,
    max_epochs=200,
    verbose=1,
    )

Note - the use_label_encoder=True, regression=False and output_num_units=20. 

My input is 2500 96*96 bw images. My label has a shape of (2500,). It happens to hold integers between 1 and 20. So Image classification problem. I saw the code using use_label_encoder and I understand it changes the label to a shape of (2500,20) using encoding.

Now - two things are quizzing me. One - I am expecting that by default (and looking at the code),  negative log loss will be used as the loss function. Is that right understanding ? The running stats shows val accuracy - as say 6.5%. Can you explain what that means w.r.t nll ?

When the network is trained and I do the prediction on it, the output shape is (3000,). I would have expected it to be (3000,20). Is this expected ?

**Comment by benanne** [comment]
The code you are talking about is part of @dnouri's nolearn package, not Lasagne: https://github.com/dnouri/nolearn
You may want to post the issue there instead.

**Comment by run2** [comment]
Ah! Ok - thanks will do that.

**Comment by benanne** [comment]
I'll close this, feel free to reopen if this ends up being more Lasagne-related after all :)
